recipes/triton/meta.yaml

```
@@ -0,0 +1,70 @@
+{% set version = "1.1.2" %}
+
+package:
+  name: triton
+  version: {{ version }}
+
+source:
+  url: https://github.com/openai/triton/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 44040696460f1d02a7251dd59468d7405982b55e112a9999ec759718066f7708
+  patches:
+    - patches/0001-patch-include-paths-in-setup.py.patch
+    - patches/0002-add-missing-header.patch
+    - patches/0003-properly-point-to-triton-includes.patch
+    - patches/0004-point-to-system-CUDA-instead-of-vendored-header.patch
+
+build:
+  number: 0
+  # TODO: windows support should be available from next version;
+  #       CPU-only support still under development
+  skip: true  # [win or cuda_compiler_version == "None"]
+  string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}
+    - make
+    - cmake
+  host:
```

```suggestion
  host:
    - cudatoolkit {{ cuda_compiler_version }}
```
Because the `nvcc=11.2` compiler has a run_export for `cudatoolkit>=11.2,<12` instead of `cudatoolkit=11.2` your [11.2 build has a runtime dependency on `cudatoolkit=11.6`](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=485510&view=logs&j=eb6ecf2b-0363-5c1a-baa7-2d771889d197&t=d8576026-8998-52ff-d027-1d631b0fb837&l=4296)! I think we need to specify the toolkit version so that the solver uses the correct version in the build environment and then users get the correct toolkit.

@leofang, is this the correct way to deal with this? Or should we be setting a toolkit requirement that matches the nvcc requirement? i.e. `cudatoolkit>=11.2,<12` instead of `cudatoolkit=11.2`

More reading: https://github.com/conda-forge/nvcc-feedstock/pull/71

---

That's intentional, and the PR you linked is the background.

> your 11.2 build has a runtime dependency on `cudatoolkit=11.6`

No, it's not a runtime dependence on 11.6, it's a run-constraint on >=11.2,<12 (which means that the appropriate cudatoolkit can be chosen based on the installation in the users environment, c.f. the `__cuda` virtual package).

---

Nevermind, the run export is also `cudatoolkit>=11.2,12`, so users won't have a problem, but it's weird that out build environment is choosing the `cudatoolkit=11.6` package.

---

I think it's normal that we build with cuda 11.2 and run with 11.6.